### PR TITLE
[Schedulers] Add exponential sigmas / exponential noise schedule

### PR DESCRIPTION
**Pull request by hlky**
# What does this PR do?

This PR adds [exponential sigmas](https://github.com/crowsonkb/k-diffusion/blob/686dbad0f39640ea25c8a8c6a6e56bb40eacefa2/k_diffusion/sampling.py#L26-L29). Added only to Euler for now, after initial review we will add it to other schedulers, and add the other schedulers in the linked issue in subsequent PRs.

The implementation is quite simple, we add `use_exponential_sigmas` parameter and `_convert_to_exponential` function.

Example usage:
```python
from diffusers import StableDiffusionXLPipeline, EulerDiscreteScheduler
import torch

pipeline: StableDiffusionXLPipeline = StableDiffusionXLPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    variant="fp16",
    torch_dtype=torch.float16,
)
pipeline.scheduler = EulerDiscreteScheduler.from_config(
    pipeline.scheduler.config, timestep_spacing="linspace", use_exponential_sigmas=True
)
```

Note that we set `timestep_spacing="linspace"` in addition to `use_exponential_sigmas=True`, this is because XL's scheduler config uses `timestep_spacing="leading"` by default and the intent is to match results from the webuis where these noise schedules are currently used. We may wish to override `timestep_spacing` to `linspace` when `use_exponential_sigmas=True` or add a warning.

Tested against [Forge](https://github.com/lllyasviel/stable-diffusion-webui-forge) (note that `cuda` device for generator is essential to match Forge results):
```python
generator = torch.Generator("cuda").manual_seed(49136503742430)

image = pipeline(
    prompt="enormous kirby. space background",
    num_inference_steps=20,
    guidance_scale=5.0,
    generator=generator,
).images[0]
image
```
output

and with the same settings on Forge we get the same result:

For reference with the default XL scheduler in Diffusers we get:
output2

#9490 

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
cc @asomoza @yiyixuxu

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_euler_discrete.py [hunk 1]
```diff
@@ -396,6 +402,10 @@ def set_timesteps(
                 sigmas = self._convert_to_karras(in_sigmas=sigmas, num_inference_steps=self.num_inference_steps)
                 timesteps = np.array([self._sigma_to_t(sigma, log_sigmas) for sigma in sigmas])
 
+            if self.config.use_exponential_sigmas:
```
Comment (yiyixuxu) [inline src/diffusers/schedulers/scheduling_euler_discrete.py:405]:
```suggestion
            elif self.config.use_exponential_sigmas:
```

File: src/diffusers/schedulers/scheduling_euler_discrete.py [hunk 2]
```diff
@@ -332,6 +336,8 @@ def set_timesteps(
             raise ValueError("Can only pass one of `num_inference_steps` or `timesteps` or `sigmas`.")
         if timesteps is not None and self.config.use_karras_sigmas:
             raise ValueError("Cannot set `timesteps` with `config.use_karras_sigmas = True`.")
+        if timesteps is not None and self.config.use_exponential_sigmas:
+            raise ValueError("Cannot set `timesteps` with `config.use_exponential_sigmas = True`.")
```
Comment (yiyixuxu) [inline src/diffusers/schedulers/scheduling_euler_discrete.py:340]:
```suggestion
            raise ValueError("Cannot set `timesteps` with `config.use_exponential_sigmas = True`.")
        if self.config.use_exponential_sigmas and self.config.use_karras_sigmas:
           raise ValueErrror("Cannot set both `config.use_exponential_sigmas = True` and config.use_karras_sigmas = True`") 
```

**Review by yiyixuxu — COMMENTED** [review]
thanks!

**Review by yiyixuxu — APPROVED** [review]
thanks!
